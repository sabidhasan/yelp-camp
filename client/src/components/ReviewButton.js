import React from 'react'
import PropTypes from 'prop-types'

const ReviewButton = (props) => {
  return (
    <button
      className='ReviewButton btn btn--large'
      aria-label='Write a Review'
      onClick={(event) => props.toggleReviewForm(event, true)}>
      <span
        className='ReviewButton__icon'
        role='img'
        aria-label='Review'
        aria-hidden='true'>
        ✏️
      </span>
      Write a Review
    </button>
  )
}

ReviewButton.propTypes = {
  toggleReviewForm: PropTypes.func.isRequired
}

export default ReviewButton
