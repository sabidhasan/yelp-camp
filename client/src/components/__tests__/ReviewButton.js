import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReviewButton from '../ReviewButton'


configure({ adapter: new Adapter() });

describe('<ReviewButton />', () => {
  let component;
  let props;
  const toggleReviewForm = jest.fn();

  beforeEach(() => {
    props = { toggleReviewForm }
    component = shallow(<ReviewButton { ...props } />)
  });

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Callback function called on click', () => {
    component.find('.ReviewButton').simulate('click');
    expect(toggleReviewForm).toHaveBeenCalledTimes(1);
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});
