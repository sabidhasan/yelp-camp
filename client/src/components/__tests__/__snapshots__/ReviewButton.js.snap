// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewButton /> LoginForm component snapshot matches what is expected 1`] = `
<button
  aria-label="Write a Review"
  className="ReviewButton btn btn--large"
  onClick={[Function]}
>
  <span
    aria-hidden="true"
    aria-label="Review"
    className="ReviewButton__icon"
    role="img"
  >
    ✏️
  </span>
  Write a Review
</button>
`;
